src/bundle/Resources/public/js/scripts/fieldType/ezdatetime.js

```diff
@@ -0,0 +1,80 @@
+(function (global) {
+    const SELECTOR_FIELD = '.ez-field-edit--ezdatetime';
+    const SELECTOR_INPUT = '.ez-data-source__input';
+    const EVENT_VALUE_CHANGED = 'valueChanged';
+
+    class EzDateTimeValidator extends global.eZ.BaseFieldValidator {
+        /**
+         * Validates the input
+         *
+         * @method validateInput
+         * @param {Event} event
+         * @returns {Object}
+         * @memberof EzDateTimeValidator
+         */
+        validateInput(event) {
+            const target = event.target;
+            const isRequired = target.required;
+            const isEmpty = !target.value.trim().length;
+            const label = event.target.closest(this.fieldSelector).querySelector('.ez-field-edit__label').innerHTML;
+            let isError = false;
+            let errorMessage = '';
+
+            if (isRequired && isEmpty) {
+                isError = true;
+                errorMessage = window.eZ.errors.emptyField.replace('{fieldName}', label);
+            }
+
+            return {
+                isError,
+                errorMessage
+            };
+        }
+    };
+
+    const validator = new EzDateTimeValidator({
+        classInvalid: 'is-invalid',
+        fieldSelector: SELECTOR_FIELD,
+        eventsMap: [
+            {
+                selector: `${SELECTOR_FIELD} ${SELECTOR_INPUT}`,
+                eventName: EVENT_VALUE_CHANGED,
+                callback: 'validateInput',
+                invalidStateSelectors: [SELECTOR_FIELD],
+                errorNodeSelectors: ['.ez-field-edit__label-wrapper'],
+            },
+        ],
+    });
+
+    validator.init();
+
+    const datetimeFields = [...document.querySelectorAll(SELECTOR_FIELD)];
+    const datetimeConfig = {
+        defaultDate: new Date(),
+        enableTime: true,
+        enableSeconds: true,
```

Comment:
`enableSeconds` should come from backend. In fieldType you can specify if field should use seconds.